src/simulation/MixedLJSimulation.cpp

```diff
@@ -0,0 +1,201 @@
+        gammas.insert({ key, -2 * std::pow(sig, 12) });
+    }
+
+    // Warn user about particles outside the domain
+    for (auto& particle : container.particles)
+        if (cellGrid.determineCellType(cellGrid.getIndexFromPos(particle.getX())) == CellType::Halo)
+            spdlog::warn(
+                "Generated particle at [{}, {}, {}] outside domain and inside a halo cell",
+                particle.getX()[0],
+                particle.getX()[1],
+                particle.getX()[2]);
+
+    // Warn user about halo cells that contain particles in the beginning
+    for (Position pos : allPositions)
+        for (auto haloCellIndex : cellGrid.haloCellIterator(pos))
+            if (!cellGrid.cells.at(haloCellIndex[0])
+                     .at(haloCellIndex[1])
+                     .at(haloCellIndex[2])
+                     ->getParticles()
+                     .empty())
+                spdlog::warn(
+                    "Halo cell at [{}, {}, {}] contains a particle",
+                    haloCellIndex[0],
+                    haloCellIndex[1],
+                    haloCellIndex[2]);
+
+    // Generate the distances needed for repulsion to occur
+    for (auto params : LJParams) {
+        if (repulsiveDistances.find(params.first) == repulsiveDistances.end())
+            repulsiveDistances.insert({ params.first, params.second.second * std::pow(2, 1 / 6) });
+    }
+
+    if (bcHandler.dimensionality != cellGrid.gridDimensionality) {
+        spdlog::error(
+            "Dimensionality mismatch between boundary conditions and cell grid.",
+            "Boundary conditions: {}, Cell grid: {}",
+            bcHandler.dimensionality,
+            cellGrid.gridDimensionality);
+        exit(EXIT_FAILURE);
+    }
+
+    if (n_thermostat) {
+        // Initialize thermostat
+        spdlog::info(
+            "Initializing thermostat with T_init={}K, T_target={}K, delta_T={}K with a frequency "
+            "of {}",
+            T_init,
+            T_target,
+            delta_T,
+            n_thermostat);
+        thermostat = Thermostat(T_init, T_target, delta_T, cellGrid.gridDimensionality);
+
+        // Intialize temperature
+        spdlog::info("Setting initial temperature to {} K", T_init);
+        thermostat.initializeBrownianMotion(this->container);
+    }
+    else{
+        spdlog::info("Themostat is turned off.");
+    }
+}
+
+void MixedLJSimulation::runSim()
+{
+    auto startTime = std::chrono::steady_clock::now();
+    unsigned long long particleUpdates = 0;
+    while (time < end_time) {
+        bcHandler.preUpdateBoundaryHandling(*this);
+
+        strategy.calF(*this);
+        strategy.calV(*this);
+        strategy.calX(*this);
+
+        bcHandler.postUpdateBoundaryHandling(*this);
+
+        ++iteration;
+        if (iteration % frequency == 0) {
+            writer->plotParticles(*this);
+        }
+        if (iteration % updateFrequency == 0) {
+            cellGrid.updateCells();
```

Comment:
`updateCells()` reassigns particles to their corresponding cells and you do this only every `updateFrequency` steps.
This can lead to _slightly_ wrong results. In the bigger context this probably doesn't matter since we already introduce a certain approximation through the cutoff. But be aware that your simulation might not do the exact same computations for different `updateFrequency` values.
The reason being that a particle might have already crossed a cell border, but is still officially two cells away and therefore not considered in a force calculation even though it might be within the cutoff.